modules/python/nitf/source/nitro.i

```diff
@@ -549,64 +549,70 @@
         /* TODO somehow get the NUMBITSPERPIXEL in the future */
         nitf_Uint8 **buf = NULL;
         PyObject* result = NULL;
-        int i, padded, rowSkip, colSkip, subimageSize;
-        nitf_SubWindow* newWindow = nitf_SubWindow_construct(error);
-        
+        int i, padded, rowSkip, colSkip, subimageSize, bandsMalloced;
+
+        bandsMalloced = 0;
+
         rowSkip = window->downsampler ? window->downsampler->rowSkip : 1;
         colSkip = window->downsampler ? window->downsampler->colSkip : 1;
-        subimageSize = (window->numRows/rowSkip) * (window->numCols/colSkip) * NITF_NBPP_TO_BYTES(nbpp);
-        
-        buf = (nitf_Uint8**) NITF_MALLOC(sizeof(nitf_Uint8*));
+        subimageSize = (window->numRows/rowSkip) * (window->numCols/colSkip) * nitf_ImageIO_pixelSize(reader->imageDeblocker);
+
+        buf = (nitf_Uint8**) NITF_MALLOC(sizeof(nitf_Uint8*) * window->numBands);
```

Comment:
You can avoid the bit of complexity `bandsMalloced` introduces if you zero out `buf` right at the beginning (that'll have the effect of setting all its pointers to NULL and doing a `free()` on a NULL pointer is fine.  Normally you would do that in one shot in C with `calloc()` but we don't have a `NITF_CALLOC()` macro.  For simplicity, I'd say just do a `memset()` after the `NITF_MALLOC()`.  Then you can ditch `bandsMalloced`.